### PR TITLE
cdc: replace tableID with Span in TableExecutor and agent 

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #7720

### What is changed and how it works?

* Replace TableID with tablepb.Span in TableExecutor
* Replace TableID with tablepb.Span in agent

This PR is compatible with older TiCDC, because #7723 convert table ID and span in message automatically.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?

No

##### Do you need to update user documentation, design documentation or monitoring documentation?

No

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/processor.go [hunk 1]
```diff
@@ -407,53 +410,49 @@ func (p *processor) IsRemoveTableFinished(tableID model.TableID) (model.Ts, bool
...
 		zap.String("changefeed", p.changefeedID.ID),
-		zap.Int64("tableID", tableID),
+		zap.Int64("tableID", span.TableID),
 		zap.Uint64("checkpointTs", checkpointTs))
 	return checkpointTs, true
 }
 
 // GetAllCurrentTables implements TableExecutor interface.
-func (p *processor) GetAllCurrentTables() []model.TableID {
+func (p *processor) GetAllCurrentTables() int {
```
Comment (3AceShowHand) [inline cdc/processor/processor.go:451]: Is it possible that we care about all tableID ?

How about adding a new method `GetAllTableCount` here?

File: cdc/processor/processor.go [hunk 2]
```diff
@@ -135,17 +137,17 @@ func (p *processor) AddTable(
 			zap.String("captureID", p.captureInfo.ID),
 			zap.String("namespace", p.changefeedID.Namespace),
 			zap.String("changefeed", p.changefeedID.ID),
-			zap.Int64("tableID", tableID),
+			zap.Int64("tableID", span.TableID),
```
Comment (3AceShowHand) [inline cdc/processor/processor.go:140]: It looks a lot of modification about replace the `tableID` to `span.TableID`.

How about add a local variable `tableID := span.TableID` first to reduce modifications ?

File: cdc/processor/processor.go (hunk 1 above)
Comment (overvenus) [reply cdc/processor/processor.go:451]: I checked code base, no one needs tableID. I will rename the method.

File: cdc/processor/processor.go (hunk 2 above)
Comment (overvenus) [reply cdc/processor/processor.go:140]: The log field will be replaced `zap.Stringer("span", span)` in later PR. I think it's okay for now, and ease my work in later PR (just search and replace).

File: cdc/scheduler/internal/table_executor.go [hunk 3]
```diff
@@ -30,33 +30,33 @@ type TableExecutor interface {
...
+		ctx context.Context, span tablepb.Span, startTs model.Ts, isPrepare bool,
 	) (done bool, err error)
 
 	// IsAddTableFinished make sure the requested table is in the proper status
-	IsAddTableFinished(tableID model.TableID, isPrepare bool) (done bool)
+	IsAddTableFinished(span tablepb.Span, isPrepare bool) (done bool)
 
 	// RemoveTable remove the table, return true if the table is already removed
-	RemoveTable(tableID model.TableID) (done bool)
+	RemoveTable(span tablepb.Span) (done bool)
```
Comment (sdojjy) [inline cdc/scheduler/internal/table_executor.go:40]: rename the function name?  it's span related func now

Comment (overvenus) [reply cdc/scheduler/internal/table_executor.go:40]: Good point, thanks!

**Comment by overvenus** [comment]
kafka tests timeout, but there is no ERROR or WARN log from schedulerv3.

**Pull request merged**